plugins/woocommerce/src/Internal/FraudProtection/SessionClearanceManager.php

```diff
@@ -0,0 +1,223 @@
+	 *
+	 * @return void
+	 */
+	public function challenge_session(): void {
+		$this->set_session_status( self::STATUS_PENDING );
+		$this->log_session_update_event( 'challenged' );
+	}
+
+	/**
+	 * Mark the current session as blocked.
+	 *
+	 * @return void
+	 */
+	public function block_session(): void {
+		$this->set_session_status( self::STATUS_BLOCKED );
+		$this->log_session_update_event( 'blocked' );
+		$this->empty_cart();
+	}
+
+	/**
+	 * Get the current session clearance status.
+	 *
+	 * @return string One of: pending, allowed, blocked.
+	 */
+	public function get_session_status(): string {
+		if ( ! $this->is_session_available() ) {
+			return self::DEFAULT_STATUS;
+		}
+
+		$status = WC()->session->get( self::SESSION_KEY, self::DEFAULT_STATUS );
+
+		// Validate status value - return default for invalid values.
+		if ( ! in_array( $status, array( self::STATUS_PENDING, self::STATUS_ALLOWED, self::STATUS_BLOCKED ), true ) ) {
+			return self::DEFAULT_STATUS;
+		}
+
+		return $status;
+	}
+
+	/**
+	 * Set the session clearance status.
+	 *
+	 * @param string $status One of: pending, allowed, blocked.
+	 * @return void
+	 */
+	private function set_session_status( string $status ): void {
+		if ( ! $this->is_session_available() ) {
+			return;
+		}
+
+		WC()->session->set( self::SESSION_KEY, $status );
+
+		// Ensure session cookie is set so the session persists across page loads.
+		// This is important because fraud protection may set session status before
+		// any cart action triggers the cookie to be set.
+		if ( WC()->session instanceof \WC_Session_Handler ) {
+			WC()->session->set_customer_session_cookie( true );
+		}
+	}
+
+	/**
+	 * Reset the session clearance status to default (allowed).
+	 *
+	 * @return void
+	 */
+	public function reset_session(): void {
+		$this->set_session_status( self::DEFAULT_STATUS );
+	}
+
+	/**
+	 * Ensure cart and session are available.
+	 *
+	 * Loads cart if not already loaded, which initializes session for both
+	 * traditional (cookie) and Store API (token) flows.
+	 *
+	 * @return void
+	 */
+	private function ensure_cart_loaded(): void {
+		if ( ! did_action( 'woocommerce_load_cart_from_session' ) && function_exists( 'wc_load_cart' ) ) {
+			WC()->call_function( 'wc_load_cart' );
+		}
+	}
+
+	/**
+	 * Check if WooCommerce session is available.
+	 *
+	 * @return bool True if session is available.
+	 */
+	private function is_session_available(): bool {
+		$this->ensure_cart_loaded();
+		return WC()->session instanceof \WC_Session;
+	}
+
+	/**
+	 * Get a unique identifier for the current session.
+	 *
+	 * @return string Session identifier.
+	 */
+	private function get_session_id(): string {
+		if ( ! $this->is_session_available() ) {
+			return 'no-session';
+		}
+
+		// Use WooCommerce session customer ID.
+		$customer_id = WC()->session->get_customer_id();
+		return $customer_id ? $customer_id : WC()->call_function( 'wc_rand_hash', 'guest_', 30 );
```

Comment:
I think this comment is similar to https://github.com/woocommerce/woocommerce/pull/62522/files#r2631888981, but in the defensive coding check we're not persisting the randomly generated guest ID, which could lead to the same session getting multiple IDs in its lifecycle.

I'm not fully familiar with what could lead the session to have no ID, but what do you think of storing the generated `guest_` ID so it can be reused later?